internal/entities/custom_rules/custom_rule.go

```diff
@@ -85,7 +95,71 @@ func (c *CustomRule) GetExpressions() (expressions []*regexp.Regexp) {
 	return expressions
 }
 
-func (c *CustomRule) ToString() string {
+func (c *CustomRule) String() string {
 	bytes, _ := json.Marshal(c)
 	return string(bytes)
 }
+
+// ruleIDValidator implements validation.Rule interface.
+type ruleIDValidator struct {
+	language languages.Language
+}
+
+// Validate implements validation.Rule.Validate.
+// nolint:funlen,exhaustive,gocyclo
+func (r ruleIDValidator) Validate(value interface{}) error {
+	id, ok := value.(string)
+	if !ok {
+		return fmt.Errorf("must be a string")
+	}
+
+	language := strings.ToUpper(r.language.ToString())
+	// TODO(matheus): Remove this. This is a terrible hack to convert C# to CSHARP.
+	// The C# rules id use the prefix CHSARP but the enum value language.CSharp is
+	// C#, so we need to convert to CSHARP nomenclature to follow already existed rules.
+	if r.language == languages.CSharp {
```

Comment:
Shouldn't we use C# prefix on rules instead of changing the language enum?

Reply:
Yeah, but on #627  we decided to use CSHARP instead C#. What do you guys think? @wiliansilvazup @nathanmartinszup

Reply:
I believe `CSHARP` will be better. We already use like this throughout all the Horusec